scripts/rnnlm/get_unigram_probs.py

```diff
@@ -130,8 +130,9 @@ def get_counts(data_sources, data_weights, vocab):
 
         with open(counts_file, 'r', encoding="utf-8") as f:
             for line in f:
-                fields = line.split()
-                assert len(fields) == 2
+                fields = line.split(' ')
```

Comment:
This was definitely a bug.  The rule for things like words.txt in Kaldi is that they should never contain anything which, *when interpreted as an ASCII character*, is a space.
And in general, we don't even assume that things like text files are actually encoded in UTF-8 or a compatible encoding; we only require that the spaces between words be ASCII space (' ').
So I believe the correct fix here would be to change encoding="utf-8" to encoding="latin-1".
Would you mind testing whether that change works for your setup?
Make sure there are no similar issues in the rnnlm/ subdirectory.

Reply:
Hi, okay, so, if the only requirement is to contain ASCII-space ' ' as a separator, this is also fine for texts in UTF encoding. The code for whitespace is always '0x20'. 

If we changed encoding in python script `encoding="utf-8"` to `encoding="latin-1"`, the further development of the python scripts would become more difficult, as the prints would not be able to print the correct symbols. So I'd keep the encoding filter as it is.

The problem was that the `line.split()` splits string with on any white-space character, while with `line.split(' ')` we are narrowing to split only with the ASCII-space `0x32`, and it does not matter if the `line` is a UTF string.

So, I'll look for other `split()` commants in the directory and change them if necessary...